worldguard-legacy/src/main/java/com/sk89q/worldguard/protection/flags/DefaultFlag.java

```diff
@@ -179,4 +189,36 @@ private DefaultFlag() {
 
         return null;
     }
+
+    /**
+     * Add a new custom flag to the flags list
+     *
+     * @param flag The flag to add
+     */
+    static void addFlag(Flag<?> flag) {
+        Flag<?> match = fuzzyMatchFlag(flag.getName());
+        if (match != null) {
+            throw new IllegalArgumentException("Duplicate flag");
+        }
+        Flag<?>[] newList = Arrays.copyOf(flagsList, flagsList.length + 1);
+        newList[flagsList.length] = flag;
+
+        // Force update the flagsList
+        // flagsList has to be public to allow other packages to access it, yet
+        // we have to be able to change it too :/
+        try {
+            java.lang.reflect.Field field = DefaultFlag.class.getField("flagsList");
+            if (!addFlagsUnlocked) {
+                addFlagsUnlocked = true;
+                java.lang.reflect.Field modifiersField = java.lang.reflect.Field.class.getDeclaredField("modifiers");
+                modifiersField.setAccessible(true);
+                modifiersField.setInt(field, field.getModifiers() & ~java.lang.reflect.Modifier.FINAL);
+            }
+            field.set(null, newList);
```

Comment:
Yes. It might be a bit strange, but it seemed to be the best way to ensure full compatibility. If I made the field writeable, other plugins could change the flags at runtime and cause breakage. If I made it private and provided a gitFlags method, I would break about all plugins out there

The way it is now, no other plugin can change the flags. At the very least, they would have to use reflection to change it, such as WGCustomFlags. I had issues with WGCustomFlags not working, so I made this quick patch.

The addFlags method is package private, so it is not exposed to other plugins. The only time you can update the flags is when the server first loads via the callback.